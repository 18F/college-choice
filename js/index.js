(function(exports) {

  var form = new formdb.Form('#search-form');
  picc.form.minifyQueryString(form);

})(this);
