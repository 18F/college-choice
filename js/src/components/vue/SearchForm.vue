<style lang="scss" scoped>
  .search-form-degree-wrapper{
    
    .v-messages{
      display: none;
    }

  }
  .subhead-2{
    margin-top: 1rem !important;
    margin-bottom: 0.3rem !important;
    font-weight: 600;
    font-family: 'Montserrat', sans-serif !important;
  }
</style>

<template>

  <v-form>
    <div class='pa-5'>
    <p class='subhead-2'>School Name</p> 
    <name-autocomplate></name-autocomplate>

    <p class='subhead-2'>Location</p>    
    <v-select v-model="input.state"
      :items="site.data.states"
      item-text="name"
      item-value="abbr"
      multiple
      chips
      hide-details
      placeholder="Select one..."
      ></v-select>
    
    <p class='subhead-2'>Field of Study/Major</p>
    <field-autocomplete v-model="input.cip4"></field-autocomplete>

    <p class='subhead-2'>Length</p>
    <div class="search-form-degree-wrapper">
        <v-checkbox
          class="search-form-degree-cb"
          v-model="input.degree"
          label="Two Year"
          value="a"
          color="primary"          
          hide-details
        ></v-checkbox>

        <v-checkbox
          class="search-form-degree-cb"
          v-model="input.degree"
          label="Four Year"
          value="b"
          color="primary"
          hide-details
        ></v-checkbox>

        <v-checkbox
          class="search-form-degree-cb"
          v-model="input.degree"
          label="Less than Two Year"
          value="c"
          color="primary"
          hide-details
        ></v-checkbox>
    </div>

    <check-range legend-title="Graduation Rate" v-model="input.completion_rate"
      :enable="utility.enable.completion_rate" @slider-toggle="utility.enable.completion_rate = $event"
      :min="0" :max="100" append-icon="mdi-percent"
    ></check-range>

    <check-range legend-title="Average Annual Cost" v-model="input.avg_net_price"
      :enable="utility.enable.avg_net_price" @slider-toggle="utility.enable.avg_net_price = $event"
      :min="0" :max="100"
    >
      <template v-slot:append-text>
        K
      </template>
    </check-range>
    </div>

    <v-expansion-panels v-if="displayAllFilters">
      <v-expansion-panel>
        <v-expansion-panel-header style='background-color: #ccc; border-radius: 0px !important;'>
          More
        </v-expansion-panel-header>
        <v-expansion-panel-content>
     
          <p class='title mt-3'>Admittance</p>
          <!-- TODO - These are not working yet -->
          <check-range v-model="input.sat"
            :enable="utility.enable.sat" 
            @slider-toggle="utility.enable.sat = $event"
            :min="0" 
            :max="1600"
            :step="100"
          >
            <template v-slot:label>
              Composite SAT
            </template>
          </check-range>

          <check-range v-model="input.act"
            :enable="utility.enable.act" @slider-toggle="utility.enable.act = $event"
            :min="0" 
            :max="36"
            :step="1"
          >
            <template v-slot:label>
              ACT Score
            </template>
          </check-range>

          <check-range v-model="input.acceptance"
            :enable="utility.enable.acceptance" 
            @slider-toggle="utility.enable.acceptance = $event"
            :min="0"
            :max="100"
            :step="5"
          >
            <template v-slot:label>
              Acceptance Rate (%)
            </template>
          </check-range>
    



        <p class='title'>School Characteristics</p>
      
          <p class='subtitle-2'>Size</p>
          <div>
          <v-btn small :class="{primary: input.size === 'small' }" @click="handleSizeClick('small')">Small</v-btn>
          <v-btn small :class="{primary: input.size === 'medium' }" @click="handleSizeClick('medium')">Medium</v-btn>
          <v-btn small :class="{primary: input.size === 'large' }" @click="handleSizeClick('large')">Large</v-btn>
          </div>
              
    
          <p class='subtitle-2 mt-5'>Type of School</p>
          <div class="search-form-type-container">
            <v-checkbox
              hide-details
              v-model="input.control"
              label="Public"
              value="public"
            ></v-checkbox>

            <v-checkbox
              hide-details
              v-model="input.control"
              label="Private Nonprofit"
              value="private"
            ></v-checkbox>

            <v-checkbox
              hide-details
              v-model="input.control"
              label="Private For-Profit"
              value="profit"
            ></v-checkbox>
          </div>

          <p class='subtitle-2 mt-5'>Urbancity</p>
          <!-- TODO - Not working yet -->
          
              <v-checkbox
                hide-details            
                v-model="input.urban"
                label="City"
                value="city"
              ></v-checkbox>

              <v-checkbox
                hide-details
                v-model="input.urban"
                label="Suburban"
                value="suburban"
              ></v-checkbox>
              <v-checkbox
                hide-details
                v-model="input.urban"
                label="Town"
                value="town"
              ></v-checkbox>

              <v-checkbox
                hide-details
                v-model="input.urban"
                label="Rural"
                value="rural"
              ></v-checkbox>
 
        <p class='subtitle-2 pt-5'>Specialized Mission</p>
        <v-select v-model='input.serving'
          :items='cleanSpecializedMission'
          item-text="value"
          item-value="key"
          placeholder='Select one...'
        ></v-select>


        <p class='subtitle-2 pt-5'>Religious Affiliation</p>
        <v-select v-model='input.religious'
          :items='site.data.religious_affiliations'
          item-text='label'
          item-value='value'
          placeholder='Select one...'
        ></v-select>


     
        </v-expansion-panel-content>
      </v-expansion-panel>
    </v-expansion-panels>

    <div id="search-submit-container" class="pa-2 text-center" v-if="!autoSubmit">
      <v-btn color="secondary" rounded @click="$emit('search-query',cleanInput)">Submit</v-btn>
    </div>

  </v-form>
</template>

<script>
import _ from 'lodash';
import querystring from 'querystring';
import CheckRange from './CheckRange.vue';
import NameAutocomplete from './NameAutocomplete.vue';
import FieldAutocomplete from './FieldAutocomplete.vue';
import { SiteData } from '../../vue/mixins/SiteData.js';

export default {
  mixins:[SiteData],
  props:{
    urlParsedParams: {
      type: Object
    },
    autoSubmit:{
      type: Boolean,
      default: false
    },
    displayAllFilters: {
      type: Boolean,
      default: false
    }
  },
  components:{
    'check-range': CheckRange,
    'name-autocomplate': NameAutocomplete,
    'field-autocomplete': FieldAutocomplete
  },
  data(){
    return{
      input:{
        state:[],
        degree:"",
        major:"",
        region:[],
        zip:"",
        distance:"",
        size:"",
        name:"",
        control:[], //Type
        serving:"",
        religious:"",
        completion_rate: null,
        avg_net_price: null,
        urban:[],
        cip4: "",
        act: null,
        sat: null,
        acceptance:null
        // page:0,
        // sort:""
      },
      utility:{
        // Hold Default state of form data.
        formDefult:{},
        // Helper to activate debounced query after initial load.
        initialized: false,
        showMore: false,
        enable:{
          completion_rate: false,
          avg_net_price: false,
          sat: false,
          act: false,
          acceptance: false
        },
      }
    }
  },
  watch:{
    cleanInput: {
      handler(newValue,oldValue){
        // On first load trigger query immediately, then debounce additional queries.
        if(!this.autoSubmit){
          return;
        }

        if(this.utility.initialized){
          this.debounceEmitSearchQuery();
        }else{
          this.$emit('search-query', newValue);
          this.utility.initialized = true;
        }
      },
      deep: true
    },
    urlParsedParams: {
      handler(newValue,oldValue){
        this.mapInputFromProp();
      },
      deep: true
    }
  },
  computed:{
    // Remove items that are not set
    cleanInput(){
      let defaultValues = this.utility.formDefult;
      // Pick only values that are different from default state.
      let groomedInput =  _.pickBy(this.input,(value,key) => {
        // If it does not exist in the default state object, remove.
        if(!_.has(defaultValues,key)){
          return false;
        }

        //If the input value is not eqaul to default, return value.
        if(!_.isEqual(value,defaultValues[key])){
          return value;
        }
      });

      // Pefrom Input to API data alterations.
      // TODO - Refactor this process. Ingest and egress.  Maybe arrary of objects with string numeral parsing.  Refactor when we have some additional time.

      // Completion rate
      if(groomedInput.completion_rate && groomedInput.completion_rate > 0 && this.utility.enable.completion_rate){
        groomedInput.completion_rate = groomedInput.completion_rate / 100 + '..';
      }else{
        _.unset(groomedInput,'completion_rate'); // TODO: CONST;
      }

      if(groomedInput.avg_net_price && groomedInput.avg_net_price > 0 && this.utility.enable.avg_net_price){
        groomedInput.avg_net_price = '..' + groomedInput.avg_net_price * 1000;
      }else{
        _.unset(groomedInput,'avg_net_price'); // TODO: CONST;
      }

      if(groomedInput.sat && groomedInput.sat > 0 && this.utility.enable.sat){
        groomedInput.sat = '..' + groomedInput.sat
      }else{
        _.unset(groomedInput,'sat'); // TODO: CONST;
      }

      if(groomedInput.act && groomedInput.act > 0 && this.utility.enable.act){
        groomedInput.act = '..' + groomedInput.act
      }else{
        _.unset(groomedInput,'act'); // TODO: CONST;
      }

      if(groomedInput.acceptance && groomedInput.acceptance > 0 && this.utility.enable.acceptance){
        groomedInput.acceptance = groomedInput.acceptance / 100 + '..'
      }else{
        _.unset(groomedInput,'acceptance'); // TODO: CONST;
      }


      return groomedInput;
    },
    // Generate a URI string of params for forwarding to search page.
    searchURL(){
      let qs = querystring.stringify(this.cleanInput);
      qs = '?' + qs.replace(/^&+/, '')
        .replace(/&{2,}/g, '&')
        .replace(/%3A/g, ':');
      
      return qs;
    },
    cleanSpecializedMission(){
      return _.map(this.site.data.special_designations,(value,key) => {
        return {
          'key': key,
          'value': value
        }
      })
    }
  },
  created(){
    // Replicate default form state.
    this.utility.formDefult = _.cloneDeep(this.input);

    // TODO - Refactor this aswell
      // For example, percentages for grad rate.
    this.mapInputFromProp();

    this.debounceEmitSearchQuery = _.debounce(function() {
      this.$emit('search-query', this.cleanInput);
    }, 1000);

  },
  mounted(){

  },
  methods:{
    mapInputFromProp(){
      this.input = _.mergeWith(this.input,this.urlParsedParams,function(objVal,newObjValue,key){
        if(_.isArray(objVal) && _.isString(newObjValue)){
          return [newObjValue];
        }
        
        // TODO - Are there consts?, add inarray check.
        // Perform any URL -> Form data translations
        if(key === 'completion_rate' || key === 'acceptance'){
          return parseFloat(newObjValue) * 100;
        }

        if(key === 'avg_net_price'){
          if(parseFloat(newObjValue.substr(2)) > 1000)
          {
            return parseFloat(newObjValue.substr(2)) / 1000;
          }
        }

        if(key === 'sat' || key === 'act'){
          return parseFloat(newObjValue.substr(2))
        }

      });

      // TODO - Refactor to a more elegant. Loop through all utility enables, and trigger on.
      // Set Sliders as active:
      if(this.input.completion_rate > 0){
        this.utility.enable.completion_rate = true;
      }

      if(this.input.avg_net_price > 0){
        this.utility.enable.avg_net_price = true;
      }

      if(this.input.sat > 0){
        this.utility.enable.sat = true;
      }

      if(this.input.act > 0){
        this.utility.enable.act = true;
      }

      if(this.input.acceptance > 0){
        this.utility.enable.acceptance = true;
      }

    },
    processChangeEvent(){
    },
    handleSizeClick(value){
      if(value === this.input.size){
        this.input.size = ""
      }else{
        this.input.size = value;
      }
    }
  }
}
</script>