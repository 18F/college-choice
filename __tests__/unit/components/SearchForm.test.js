import {shallowMountWithConfig, mountWithConfig} from '../../utility/MountWithConfig.js';

import SearchForm from '../../../js/src/components/vue/SearchForm.vue';

// Ingest of params with proper mapped state
// Testing clean input computed property
// Epected Output


describe('SearchForm.vue', () => {

  beforeEach(() => {
  })
  
  test('Renders correctly', async () => {
    const mountOptions = {
      propsData:{
        displayAllFilters: true,
        autoSubmit: true,
        urlParsedParams:{}
      },
    }

    const wrapper = shallowMountWithConfig(SearchForm,mountOptions);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is a Vue instance', () => {
    const mountOptions = {
      propsData:{
        displayAllFilters: true,
        autoSubmit: true,
        urlParsedParams:{}
      },
      methods:{
        debounceEmitSearchQuery(){
          this.$emit('search-query', "Test");
        }
      }
    }

    const wrapper = shallowMountWithConfig(SearchForm,mountOptions);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

})