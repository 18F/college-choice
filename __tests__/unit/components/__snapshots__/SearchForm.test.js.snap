// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm.vue Renders correctly 1`] = `
<v-form-stub>
  <div
    class="px-4 pt-2 pb-4 grey lighten-3"
  >
    <h4
      class="subhead-2 font-weight-bold mb-2"
    >
      Search by Name
    </h4>
     
    <name-autocomplete-stub
      initial_school=""
    />
  </div>
   
  <div
    class="py-2 px-5"
  >
    <p
      class="subhead-2"
      id="location-label"
    >
      
    Location
  
    </p>
     
    <v-select-stub
      appendicon="$dropdown"
      aria-labelledby="location-label"
      backgroundcolor=""
      class="mb-3 mt-0 pt-0"
      clearicon="$clear"
      errorcount="1"
      errormessages=""
      hidedetails="true"
      id="search-from-location-select"
      itemcolor="primary"
      itemdisabled="disabled"
      items="Near Me,ZIP Code,State"
      itemtext="text"
      itemvalue="value"
      loaderheight="2"
      menuprops="[object Object]"
      messages=""
      nodatatext="$vuetify.noDataText"
      placeholder="Select an option"
      rules=""
      successmessages=""
      type="text"
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <!---->
     
    <!---->
     
    <!---->
     
    <p
      class="subhead-2"
    >
      Field of Study Offered 
      <tooltip-stub
        color="blue darken-3"
        definition="field-of-study"
        version="default"
      />
    </p>
     
    <div
      class="mt-4 pl-4 ml-2"
      id="search-form-sub-degree-container"
    >
      <p
        class="subhead-2"
        id="fields-label"
      >
        Academic Fields
      </p>
       
      <field-autocomplete-stub
        value=""
      />
       
      <div>
        <p
          class="subhead-2"
        >
          Degrees/Certificate
        </p>
         
        <v-checkbox-stub
          backgroundcolor=""
          class="search-form-degree-cb my-0 py-0"
          color="secondary"
          errorcount="1"
          errormessages=""
          hidedetails="true"
          id="search-form-fos-degree-c"
          indeterminateicon="$checkboxIndeterminate"
          label="Certificate"
          messages=""
          officon="$checkboxOff"
          onicon="$checkboxOn"
          ripple="true"
          rules=""
          successmessages=""
          value=""
          valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
        />
         
        <v-checkbox-stub
          backgroundcolor=""
          class="search-form-degree-cb my-0 py-0"
          color="secondary"
          errorcount="1"
          errormessages=""
          hidedetails="true"
          id="search-form-fos-degree-a"
          indeterminateicon="$checkboxIndeterminate"
          label="Associate's Degree"
          messages=""
          officon="$checkboxOff"
          onicon="$checkboxOn"
          ripple="true"
          rules=""
          successmessages=""
          value=""
          valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
        />
         
        <v-checkbox-stub
          backgroundcolor=""
          class="search-form-degree-cb my-0 py-0"
          color="secondary"
          errorcount="1"
          errormessages=""
          hidedetails="true"
          id="search-form-fos-degree-b"
          indeterminateicon="$checkboxIndeterminate"
          label="Bachelor's Degree"
          messages=""
          officon="$checkboxOff"
          onicon="$checkboxOn"
          ripple="true"
          rules=""
          successmessages=""
          value=""
          valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
        />
      </div>
    </div>
     
    <check-range-stub
      andup="true"
      appendtext="%"
      class="mt-5"
      id="search-form-completion-rate"
      legendtitle="Graduation Rate"
      max="100"
      min="0"
      step="5"
    />
     
    <check-range-stub
      appendtext="k"
      id="search-form-avg-net-price"
      legendtitle="Average Annual Cost"
      max="100"
      min="0"
      step="5"
    />
  </div>
   
  <div
    class="px-5"
  >
    <check-range-stub
      id="search-form-sat-math"
      legendtitle="Title"
      max="800"
      min="0"
      step="20"
    />
     
    <check-range-stub
      id="search-form-sat-read"
      legendtitle="Title"
      max="800"
      min="0"
      step="20"
    />
     
    <check-range-stub
      id="search-form-act"
      legendtitle="Title"
      max="36"
      min="0"
      step="1"
    />
     
    <check-range-stub
      andup="true"
      appendtext="%"
      id="search-form-acceptance"
      legendtitle="Title"
      max="100"
      min="0"
      step="5"
    />
     
    <p
      class="subhead-2 mb-3"
    >
      Size
    </p>
     
    <div>
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-size-small"
        indeterminateicon="$checkboxIndeterminate"
        label="Small"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
       
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-size-medium"
        indeterminateicon="$checkboxIndeterminate"
        label="Medium"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
       
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-size-large"
        indeterminateicon="$checkboxIndeterminate"
        label="Large"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
    </div>
     
    <p
      class="subhead-2"
    >
      Type of School
    </p>
     
    <div
      class="search-form-type-container"
    >
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-type-public"
        indeterminateicon="$checkboxIndeterminate"
        label="Public"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
       
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-type-private"
        indeterminateicon="$checkboxIndeterminate"
        label="Private Nonprofit"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
       
      <v-checkbox-stub
        backgroundcolor=""
        class="py-0 my-0"
        color="secondary"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-form-type-profit"
        indeterminateicon="$checkboxIndeterminate"
        label="Private For-Profit"
        messages=""
        officon="$checkboxOff"
        onicon="$checkboxOn"
        ripple="true"
        rules=""
        successmessages=""
        value=""
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
    </div>
     
    <p
      class="subhead-2"
    >
      Urbanicity
    </p>
     
    <v-checkbox-stub
      backgroundcolor=""
      class="py-0 my-0"
      color="secondary"
      errorcount="1"
      errormessages=""
      hidedetails="true"
      id="search-form-locale-city"
      indeterminateicon="$checkboxIndeterminate"
      label="City"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <v-checkbox-stub
      backgroundcolor=""
      class="py-0 my-0"
      color="secondary"
      errorcount="1"
      errormessages=""
      hidedetails="true"
      id="search-form-locale-suburban"
      indeterminateicon="$checkboxIndeterminate"
      label="Suburban"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <v-checkbox-stub
      backgroundcolor=""
      class="py-0 my-0"
      color="secondary"
      errorcount="1"
      errormessages=""
      hidedetails="true"
      id="search-form-locale-town"
      indeterminateicon="$checkboxIndeterminate"
      label="Town"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <v-checkbox-stub
      backgroundcolor=""
      class="py-0 my-0"
      color="secondary"
      errorcount="1"
      errormessages=""
      hidedetails="true"
      id="search-form-locale-rural"
      indeterminateicon="$checkboxIndeterminate"
      label="Rural"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <p
      class="subhead-2"
      id="specialized-mission-label"
    >
      Specialized Mission
    </p>
     
    <v-select-stub
      appendicon="$dropdown"
      aria-labelledby="specialized-mission-label"
      backgroundcolor=""
      class="py-0 my-0"
      clearable="true"
      clearicon="$clear"
      color="secondary"
      errorcount="1"
      errormessages=""
      id="search-form-serving"
      itemcolor="primary"
      itemdisabled="disabled"
      items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      itemtext="value"
      itemvalue="key"
      loaderheight="2"
      menuprops="[object Object]"
      messages=""
      nodatatext="$vuetify.noDataText"
      placeholder="Select one..."
      rules=""
      successmessages=""
      type="text"
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
     
    <p
      class="subhead-2"
      id="religions-affiliation-label"
    >
      Religious Affiliation
    </p>
     
    <v-select-stub
      appendicon="$dropdown"
      aria-labelledby="religions-affiliation-label"
      backgroundcolor=""
      class="py-0 my-0"
      clearable="true"
      clearicon="$clear"
      color="secondary"
      errorcount="1"
      errormessages=""
      id="search-form-religous"
      itemcolor="primary"
      itemdisabled="disabled"
      items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      itemtext="label"
      itemvalue="value"
      loaderheight="2"
      menuprops="[object Object]"
      messages=""
      nodatatext="$vuetify.noDataText"
      placeholder="Select one..."
      rules=""
      successmessages=""
      type="text"
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
  </div>
   
  <!---->
</v-form-stub>
`;
